test/ibmq/test_proxies.py

```diff
@@ -12,83 +12,112 @@
+"""Tests for the IBMQClient and IBMQVersionFinder proxy support."""
 
 import urllib
 import subprocess
 
 from requests.exceptions import ProxyError
 
-from qiskit.providers.ibmq.api_v2 import IBMQClient
+from qiskit.providers.ibmq.api_v2 import IBMQClient, IBMQVersionFinder
 from qiskit.providers.ibmq.api_v2.exceptions import RequestsApiError
 from qiskit.test import QiskitTestCase, requires_qe_access
 
 from ..decorators import requires_new_api_auth
 
 ADDRESS = '127.0.0.1'
 PORT = '8080'
+VALID_PROXIES = {'https': 'http://{}:{}'.format(ADDRESS, PORT)}
+INVALID_PORT_PROXIES = {'https': '{}:{}'.format(ADDRESS, '6666')}
+INVALID_ADDRESS_PROXIES = {'https': '{}:{}'.format('invalid', PORT)}
 
 
 class TestProxies(QiskitTestCase):
     """Tests for proxy capabilities."""
 
-    @classmethod
-    def setUpClass(cls):
-        super().setUpClass()
-
-        # Launch the mock server.
-        cls.proxy_process = subprocess.Popen([
+    def setUp(self):
+        # launch a mock server.
+        self.proxy_process = subprocess.Popen([
             'pproxy', '-v', '-i', 'http://{}:{}'.format(ADDRESS, PORT)
         ], stdout=subprocess.PIPE)
 
-    @classmethod
-    def tearDownClass(cls):
-        super().tearDownClass()
-
-        # Close the mock server.
-        if cls.proxy_process.returncode is None:
-            cls.proxy_process.terminate()
+    def tearDown(self):
+        # terminate the mock server.
+        if self.proxy_process.returncode is None:
+            self.proxy_process.terminate()
 
     @requires_qe_access
     @requires_new_api_auth
-    def test_proxies(self, qe_token, qe_url):
-        """Should reach the proxy."""
-        input_proxies = {
-            'https': 'http://{}:{}'.format(ADDRESS, PORT)
-        }
+    def test_proxies_ibmqclient(self, qe_token, qe_url):
+        """Should reach the proxy using IBMQClient."""
+        pproxy_desired_access_log_line_ = pproxy_desired_access_log_line(qe_url)
+
+        _ = IBMQClient(qe_token, qe_url, VALID_PROXIES)
+
+        self.proxy_process.terminate()  # kill to be able of reading the output
+        self.assertIn(pproxy_desired_access_log_line_,
+                      self.proxy_process.stdout.read().decode('utf-8'))
 
-        qe_url_parts = urllib.parse.urlparse(qe_url)
-        protocol_port = '443' if qe_url_parts.scheme == 'https' else '80'
-        pproxy_desired_access_log_line = 'http {}:{}'.format(qe_url_parts.hostname, protocol_port)
+    # pylint: disable=unused-argument
+    @requires_qe_access
+    @requires_new_api_auth
+    def test_proxies_ibmqversionfinder(self, qe_token, qe_url):
+        """Should reach the proxy using IBMQVersionFinder."""
+        pproxy_desired_access_log_line_ = pproxy_desired_access_log_line(qe_url)
 
-        _ = IBMQClient(qe_token, qe_url, input_proxies)
+        _ = IBMQVersionFinder(qe_url, VALID_PROXIES)
+        _.version()  # call the version finder, sending logging output to the proxy process
```

Comment:
In general we use `_` as the name of a variable only in the cases where that variable is not going to be used anymore - and in this case, it is being used. Can you rename it to a sensible name?

Reply:
(and also in `test_invalid_proxy_port_ibmqversionfinder` and `test_invalid_proxy_address_ibmqversionfinder`)